OWML.ModHelper/ModConfig.cs

```diff
@@ -65,5 +65,57 @@ public T GetSetting<T>(string key)
             return GetSettingsValue<T>(key);
         }
 
+        public void MakeConfigConsistentWithDefaults(IModConfig defaultConfig)
+        {
+            if (defaultConfig == null)
+            {
+                return;
+            }
+
+            AddMissingDefaults(defaultConfig);
+
+            var settingsCopy = new Dictionary<string, object>(Settings);
+            foreach (var setting in Settings)
+            {
+                if (!defaultConfig.Settings.ContainsKey(setting.Key))
+                {
+                    settingsCopy.Remove(setting.Key);
+                    continue;
+                }
+                else if (!IsSettingSameType(setting.Value, defaultConfig.Settings[setting.Key]))
+                {
+                    settingsCopy[setting.Key] = defaultConfig.Settings[setting.Key];
+                    continue;
+                }
+            }
+            Settings = settingsCopy;
+        }
+
+        private void AddMissingDefaults(IModConfig defaultConfig)
+        {
+            foreach (var defaultSetting in defaultConfig.Settings)
+            {
+                if (!Settings.ContainsKey(defaultSetting.Key))
+                {
+                    Settings.Add(defaultSetting.Key, defaultSetting.Value);
```

Comment:
pretty sure you can do something like
`Settings.Concat(defaultConfig.Settings.Where( x=> !defaultConfig.Settings.Keys.Contains(defaultSetting.Key)));`

Reply:
maybe it's Union though

Reply:
i think unions don't support dicts without a custom comparer

Reply:
pretty sure I used Union in HashSets, don't think Dictionary would be much different

Reply:
That doesn't look any better to me. Just makes it hard to read with no obvious advantage.

I know @amazingalek has a sweet spot for this kind of one-liner, so maybe he can chime in.

Reply:
BTW this code was just moved over from `ModFinder.cs` with almost no changes.

Reply:
ah, it's UnionWith rather

Reply:
https://github.com/amazingalek/owml/pull/210 🥴